sky/clouds/aws.py

```diff
@@ -856,6 +863,37 @@ def check_quota_available(cls,
         # Quota found to be greater than zero, try provisioning
         return True
 
+    def get_reservations_available_resources(
+        self,
+        instance_type: str,
+        region: str,
+        zone: Optional[str],
+        specific_reservations: Set[str],
+    ) -> Dict[str, int]:
+        if zone is None:
+            # For backward compatibility, the cluster in INIT state launched
+            # before #2352 may not have zone information. In this case, we
+            # return 0 for all reservations.
+            return {reservation: 0 for reservation in specific_reservations}
```

Comment:
Should we also filter this by the `specific_reservations`?

Reply:
Just checking, does that mean such a cluster cannot use reservations when re-launching?

Reply:
This is only for backward compatibility and only old clusters will not work with the reservation (similar to our GCP reservation support implementation)